src/sql/src/plan/statement/ddl.rs

```diff
@@ -4114,13 +4124,70 @@ pub fn plan_drop_objects(
             }),
         }
     }
-    let drop_ids = scx.catalog.object_dependents(&referenced_ids);
 
-    Ok(Plan::DropObjects(DropObjectsPlan {
+    // Check that dropping this object does not have any implicit dependencies
+    // (e.g. `TEXT COLUMNS` references for subsources).
+    for id in &referenced_ids {
```

Comment:
Right now, we ensure that all of the `TEXT COLUMNS` values refer to columns present in the publication tables. 

Adding subsources now drops tables from the publication if they are no longer referred to.

To prevent `TEXT COLUMNS` from referring to tables that we remove from the publication, I figured this was the easiest way to keep things in sync. Ultimately, we should just remove `ALTER SOURCE...DROP SUBSOURCE` entirely and reconcile away dangling text column references when adding subsources. To minimize the complexity of the diff, I just added this rather than removing the command entirely.

However, explaining all of this makes me realize that this stop gap is more confusing than just fixing the issue. I'll remove this code, alongside `ALTER SOURCE...DROP SUBSOURCE`. I'll change the code such that adding a subsource reconciles the `TEXT COLUMNS` references. I'll also file a `TODO` to allow ad-hoc reconciliation of the `TEXT COLUMNS` references without adding a subsource.

Reply:
I resolved this by removing the code tagged here, deprecated `ALTER SOURCE...DROP SUSBOURCE`, and fixed `SHOW CREATE SOURCE` so that it always provides roundtrip-able values. We might be able to simplify the `SHOW CREATE SOURCE` implementation if we move `TEXT COLUMNS` to the `CREATE SUBSOURCE` statement, which I filed #26774 to track.